src/lib/plugins/blocks.js

```diff
@@ -60,14 +62,16 @@ module.exports.player = function (player, serv) {
       if (!results) return false
       return results
     },
-    action (params) {
+    action (params, ctx) {
       let res = params.slice(1, 4)
-      res = res.map((val, i) => serv.posFromString(val, player.position[['x', 'y', 'z'][i]]))
-      player.setBlock(new Vec3(res[0], res[1], res[2]).floored(), params[4], params[5] || 0)
+      if (ctx.player) res = res.map((val, i) => serv.posFromString(val, ctx.player.position[['x', 'y', 'z'][i]]))
+      else res = res.map((val, i) => serv.posFromString(val, new Vec3(0, 128, 0)[['x', 'y', 'z'][i]]))
```

Comment:
Why this position ? Does the vanilla server do that ?

Reply:
idk, if I type ``setblock ~1 ~1 ~1 stone``, server will return ``Cannot place block outside of the world``